### PR TITLE
FIx NumberProxy to Counter

### PR DESCRIPTION
**Pull request by dc7303**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature

/kind feature

**What this PR does / why we need it**:
We discussed in yorkie-team/yorkie-js-sdk#57 that it is inefficient for `Primitive` to get a value with `getValue()`.  
And we thought positively about introducing types like `Counter` in `Automerge`.  
So I added `Counter` and `CounterProxy` which can use `Increase operation`.

In addition, as `MetricsProt` was added to the previous commit, a duplicate error with `RPCPort` occurred. So, because an error occurred during the test, the `RPCPort` was modified.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

### CODE REVIEW

**Comment by hackerwins** [comment]
@dc7303 Thank you for your contribution. I will review this soon.

File: pkg/document/json/counter.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+ * limitations under the License.
+ */
+
+package json
+
+import (
+	"encoding/binary"
+	"fmt"
+	"github.com/yorkie-team/yorkie/pkg/document/time"
+	"math"
```
Comment (hackerwins) [inline pkg/document/json/counter.go:23]: Please run `make fmt` before committing.
https://github.com/yorkie-team/yorkie#developing-yorkie

File: pkg/document/proxy/counter_proxy.go [hunk 2]
```diff
@@ -24,61 +24,61 @@ import (
...
 }
 
-// NewNumberProxy create NumberProxy instance.
-func NewNumberProxy(ctx *change.Context, primitive *json.Primitive) *NumberProxy {
-	valueType := primitive.ValueType()
-	if valueType != json.Integer && valueType != json.Long && valueType != json.Double {
+// NewCounterProxy create CounterProxy instance.
+func NewCounterProxy(ctx *change.Context, counter *json.Counter) *CounterProxy{
+	valueType := counter.ValueType()
+	if valueType != json.IntegerCnt && valueType != json.LongCnt && valueType != json.DoubleCnt {
```
Comment (hackerwins) [inline pkg/document/proxy/counter_proxy.go:36]: Can we use `isNumericType()` here?

File: testhelper/testhelper.go [hunk 3]
```diff
@@ -35,7 +35,7 @@ import (
 var testStartedAt int64
 
 const (
-	RPCPort     = 1101
+	RPCPort     = 1100
```
Comment (hackerwins) [inline testhelper/testhelper.go:38]: I didn't understand why the RPCPort crashed with MetricsPort.

Comment (dc7303) [reply testhelper/testhelper.go:38]: https://github.com/yorkie-team/yorkie/blob/1b396353b63cd7529289a407c75fada6b89cebb1/yorkie/rpc/server_test.go#L22

There is a place to use by adding 1 to `testhelper.RPCPort`.

Comment (dc7303) [reply testhelper/testhelper.go:38]:
```bash
2020-08-30T01:19:37.096+0900    ERROR   listen tcp :1102: bind: address already in use
github.com/yorkie-team/yorkie/yorkie/rpc.(*Server).listenAndServeGRPC
        /Users/choedongcheol/Workspace/opensource/yorkie/yorkie/rpc/server.go:363
github.com/yorkie-team/yorkie/yorkie/rpc.(*Server).Start
        /Users/choedongcheol/Workspace/opensource/yorkie/yorkie/rpc/server.go:97
github.com/yorkie-team/yorkie/yorkie/rpc_test.TestMain
        /Users/choedongcheol/Workspace/opensource/yorkie/yorkie/rpc/server_test.go:47
main.main
        _testmain.go:44
runtime.main
        /Users/choedongcheol/.gvm/gos/go1.13.8/src/runtime/proc.go:203
2020/08/30 01:19:37 failed rpc listen: listen tcp :1102: bind: address already in use

```

**Comment by dc7303** [comment]
@hackerwins 
Thanks for the feedback. I modified it and pushed it.
Additionally, `NewCounter()` has been improved. Please refer to the ca0b4a3 commit message for details.

**Review by hackerwins — APPROVED** [review]
LGTM

**Pull request merged**